### PR TITLE
[JS] docs: Updated README to match new changes to OYD Sample

### PR DESCRIPTION
**Pull request by Carter425**
#minor

## Details

Updated the README to the AzureOpenAI sample with updated directions utilizing the Test Tool now that John has it working, removed unneeded dependency directions such as Yarn install and Yarn Build, and added additional changes for clarity.

#### Change details

Updated the README to the AzureOpenAI sample with updated directions utilizing the Test Tool now that John has it working, removed unneeded dependency directions such as Yarn install and Yarn Build, and added additional changes for clarity.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Carter425** [comment]
@corinagum and @aacebo, this commit updates the Azure OYD Sample README reflecting some updates John just made such as fixing the Test Tool and I also removed un-needed directions for dependencies like Yarn install and Yarn Build per Alex saying dependencies are done automatically.

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 1]
```diff
@@ -31,20 +31,15 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
 | [Visual Studio Code](https://code.visualstudio.com/download)                                                                  | Typescript build environments. Use the latest version.                                                                                                                                                                                                              |
 | [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension) (5.3.x or greater) | Microsoft Visual Studio Code extension that creates a project scaffolding for your app. Use the latest version.                                                                                                                                                     |
-| [Git](https://git-scm.com/downloads)                                                                                          | Git is a version control system that helps you manage different versions of code within a repository.                                                                                                                                                               |
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:34]: I don't think we want to remove this part. If people are cloning the repo, they will still need git.

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 2]
```diff
@@ -31,20 +31,15 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
-| [Microsoft Teams](https://www.microsoft.com/microsoft-teams/download-app)                                                     | Microsoft Teams to collaborate with everyone you work with through apps for chat, meetings, and call-all in one place.                                                                                                                                              |
-| [Microsoft&nbsp;Edge](https://www.microsoft.com/edge) (recommended) or [Google Chrome](https://www.google.com/chrome/)        | A browser with developer tools.                                                                                                                                                                                                                                     |
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:37]: Why are we removing this prerequisite?

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 3]
```diff
@@ -31,20 +31,15 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
-| [Microsoft 365 developer account](/microsoftteams/platform/concepts/build-and-test/prepare-your-o365-tenant)                  | Access to Teams account with the appropriate permissions to install an app and [enable custom Teams apps and turn on custom app uploading](../../../concepts/build-and-test/prepare-your-o365-tenant.md#enable-custom-teams-apps-and-turn-on-custom-app-uploading). |
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:38]: Same question - don't they need developer accounts and azure openai?

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 4]
```diff
@@ -31,20 +31,15 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
-| [Azure OpenAI](https://oai.azure.com/portal)                                                                                  | Deploy Azure OpenAI large language models and test your custom copilot ideas in the Azure OpenAI Studio Playground. If you want to host your app or access resources in Microsoft Azure, you must create an Azure OpenAI service.                                                                                                                                                                                                                                    |
+| [Microsoft Teams](https://www.microsoft.com/microsoft-teams/download-app)                                                     | Access to a Microsoft Teams account with the appropriate permissions to install an app, [enable custom Teams apps, and turn on custom app uploading.](https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/prepare-your-o365-tenant#enable-custom-teams-apps-and-turn-on-custom-app-uploading)                                                                                                                                             |
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:35]: They would still need a developer account right?

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 5]
```diff
@@ -55,65 +50,38 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
+1. If you chose `API key` in data connection, manually copy and paste your Azure AI Search key in `src\prompts\chat\config.json` file. Your **Azure AI Search Key** can be found in Azure OpenAI Studio Playground by clicking the `View code` button and looking under **Azure Search Resource Key**. If you chose `system assigned managed identity`, you can skip this step. Learn more about different data connection options in the [Data connection section](https://review.learn.microsoft.com/en-us/azure/ai-services/openai/concepts/use-your-data?tabs=ai-search%2Ccopilot&branch=pr-en-us-277392#data-connection) .
    
-1. Open the Visual Studio terminal and log into Azure CLI selecting the Azure account that you assigned Cognitive Service OpenAI User role to. This is required because this sample uses managed identity. Use the following command to log in:
+1. Open the Visual Studio terminal by selecting `View > Terminal` and log into Azure CLI selecting the Azure account that you assigned Cognitive Service OpenAI User role to. Use the following command to log in:
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:65]:
```suggestion
1. Open the Visual Studio Code terminal by selecting `View > Terminal` and log into Azure CLI selecting the Azure account that you assigned Cognitive Service OpenAI User role to. Use the following command to log in:
```
Code, right?

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 6]
```diff
@@ -55,65 +50,38 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
-1. Select `Debug in Teams (Edge)` or `Debug in Teams (Chrome)` from the `RUN AND DEBUG ▷` dropdown. IMPORTANT: `Debug in Test Tool` does not work with this feature at this time.
-   
-1. Select `Run > Start Debugging` (F5).
-   
-1. A browser tab opens a Teams web client requesting to add the bot to your tenant. Select Add to begin a chat with your custom copilot
-
-> If you do not have permission to upload custom apps (sideloading), Teams Toolkit will recommend creating and using a Microsoft 365 Developer Program account - a free program to get your own dev environment sandbox that includes Teams.   
-
+ 
+1. To debug your app, press the `F5` key or from the left pane, select `RUN AND DEBUG ▷` (Ctrl+Shift+D) and then select `Debug in (Test tool)` from the dropdown list.  Select `Run > Start Debugging` (F5). The Test Tool opens in a webpage where you can chat with your custom copilot.
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:70]:
```suggestion
1. To debug your app, press the `F5` key or from the left pane, select `RUN AND DEBUG ▷` (Ctrl+Shift+D) and then select `Debug in Test Tool` from the dropdown list.  Select `Run > Start Debugging` (F5). The Test Tool opens in a webpage where you can chat with your custom copilot. 
```
According to John's PR

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 7]
```diff
@@ -55,65 +50,38 @@ This guide will show you have the set up your custom copilot for Teams using Azu
...
-
+ 
+1. To debug your app, press the `F5` key or from the left pane, select `RUN AND DEBUG ▷` (Ctrl+Shift+D) and then select `Debug in (Test tool)` from the dropdown list.  Select `Run > Start Debugging` (F5). The Test Tool opens in a webpage where you can chat with your custom copilot. 
 
 ### Provisioning, Deploying, and Publishing your custom Copilot
-After you have tested it locally, you can provision, deploy and publish your Teams app using Teams Toolkit. 
+After you've tested your copilot, you can provision, deploy, and publish your Teams app by selecting the Teams Toolkit Extension on the left pane in Visual Studio. Run the separate provision, deploy, and publish stages in Lifecycle section. You may be asked to sign into your Microsoft 365 account where you have permissions to upload custom apps and your Azure Account.
 
 **IMPORTANT** As this sample uses managed identity, for your custom copilot to generate responses you must assign Cognitive Service OpenAI User role to your custom copilot’s App Service resource group after deploying your app to Azure.
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:75]: This section should go after the 'Provision your app' etc. list below

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 8]
```diff
@@ -139,6 +107,7 @@ As this sample uses managed identity, you must enable assign Cognitive Service O
 
 Now, your published custom copilot Teams app is ready for use.
 
+## Supplementary Details and Tips
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:110]: In other samples I think we use "Other resources" but this change is up to you

**Review by corinagum — CHANGES_REQUESTED** [review]
Hey Carter, thanks for the PR. I left some suggestions.

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md (hunk 2 above)
Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:37]: The Test tool should be enabled now so the browser prerequisite is not needed. The Microsoft Teams prerequisite is still there just lower down.

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md (hunk 1 above)
Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:34]: I was trying to simplify this and thought it was more of a recommendation as the Teams AI QuickStart guide did not have it. 

Reference: https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/teams%20conversational%20ai/conversation-ai-quick-start?tabs=OpenAI-key%2COpenAI-key2&pivots=qs-javascript

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md (hunk 3 above)
Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:38]: I was trying to simplify again. A developer account is not needed for the test tool, but it is for test in edge or to deploy the bot. I noted in the Microsoft Teams requirement the need for custom app enablement. I'll add this back in

Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:38]: added back in with the latest commit

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md (hunk 4 above)
Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md]: added back in with the latest commit

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md (hunk 8 above)
Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:111]: I'll keep this as is if it's acceptable. Thank you for the info

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md (hunk 2 above)
Comment (corinagum) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:37]: Instead of removing, can we say (if wanting to test with Edge)?

Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:37]: Added back in with the noting it as "Optional: a browser with developer tools if testing in Edge or Chrome is desired."

File: js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md [hunk 9]
```diff
@@ -139,12 +110,13 @@ As this sample uses managed identity, you must enable assign Cognitive Service O
...
 
 ### Enable your Custom Copilot for Group Chats and Channels
 
 This custom copilot sample is pre-configured for only personal chats (1 on 1) during preview due to ongoing testing from Azure OpenAI On Your Data to determine the effects of group chats on AI response quality. Group chats can be enabled with the understanding that the AI response quality has not been fully tested for these scenarios. 
 
-A custom copilot can be mentioned ("@customcopilotname") in a channel if it has been added to the team. Note that additional replies to a custom copilot in a channel require @ mentioning the custom copilot. It will not respond to replies where it isn't mentioned. See here for more information. 
+A custom copilot can be mentioned ("@customcopilotname") in a channel if it has been added to the team. Note that additional replies to a custom copilot in a channel require @ mentioning the custom copilot. It will not respond to replies where it isn't mentioned. See [here](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/channel-and-group-conversations?tabs=dotnet) for more information.
```
Comment (corinagum) [inline js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md:119]: Should link to typescript, not dotnet

Comment (Carter425) [reply js/samples/04.ai-apps/h.datasource-azureOpenAI/README.md]: updated

**Review by corinagum — APPROVED** [review]

**Pull request merged**